src/npcmove.cpp

```diff
@@ -3182,8 +3182,8 @@ void npc::do_reload( item &it )
 
     // Note: we may be reloading the magazine inside, not the gun itself
     // Maybe @todo: allow reload functions to understand such reloads instead of const casts
-    auto &target = const_cast<item &>( *reload_opt.target );
-    auto &usable_ammo = reload_opt.ammo;
+    item &target = const_cast<item &>( *reload_opt.target );
```

Comment:
Which means it will always be an item &, which means auto is accomplishing nothing.
The default is explicit typing, you should have a reason to use auto such as polymorphic code, that is not the case here.